visualization/app/codeCharta/ui/nodeContextMenu/nodeContextMenu.component.ts

```diff
@@ -72,6 +72,11 @@ export class NodeContextMenuController
 			event => NodeContextMenuController.broadcastHideEvent(this.$rootScope, new Vector2(event.x, event.y)),
 			true
 		)
+		document.body.addEventListener(
+			"wheel",
+			event => NodeContextMenuController.broadcastHideEvent(this.$rootScope, new Vector2(event.x, event.y)),
```

Comment:
Would it not be better to attach the event listener to the concrete element that we want to check the wheel upon? That way we would not need the vector. The vector seems like a bit of overhead in general as it's only an object with x and y and I would just pass through the current event, since it's already allocated instead of allocating a new object.

`onHideNodeContextMenu()` could also be improved (which is triggered due to the event) to accept `undefined` as input. That way there's no need to check for the bounds with a default vector of -1, -1.

Reply:
I refactored the NodeContextMenu component and applied your suggestion regarding the EventListeners.
Appropriate EventListeners will be added when the menu is shown and removed immediately after the listeners have been called. Therefore, the listeners will be called only, if the menu is actually opened.